cmd2/command_definition.py

```diff
@@ -0,0 +1,131 @@
+# coding=utf-8
+"""
+Supports the definition of commands in separate classes to be composed into cmd2.Cmd
+"""
+import functools
+from typing import (
+    Callable,
+    Dict,
+    Iterable,
+    Optional,
+    Type,
+    Union,
+)
+from .constants import COMMAND_FUNC_PREFIX
+
+# Allows IDEs to resolve types without impacting imports at runtime, breaking circular dependency issues
+try:  # pragma: no cover
+    from typing import TYPE_CHECKING
+    if TYPE_CHECKING:
+        from .cmd2 import Cmd, Statement
+        import argparse
+except ImportError:
+    pass
+
+_REGISTERED_COMMANDS = {}  # type: Dict[str, Callable]
+"""
+Registered command tuples. (command, do_ function, complete_ function, help_ function
+"""
+
+
+def _partial_passthru(func: Callable, *args, **kwargs) -> functools.partial:
+    """
+    Constructs a partial function that passes arguments through to the wrapped function.
+    Must construct a new type every time so that each wrapped function's __doc__ can be copied correctly.
+
+    :param func: wrapped function
+    :param args: positional arguments
+    :param kwargs: keyword arguments
+    :return: partial function that exposes attributes of wrapped function
+    """
+    def __getattr__(self, item):
+        return getattr(self.func, item)
+
+    def __setattr__(self, key, value):
+        return setattr(self.func, key, value)
```

Comment:
Recommend adding unit test for this method

Reply:
I only really needed the getattr to work to pass through the docstring from the wrapped function. I added setattr for completeness but I don't really know how to test this in a sensible fashion.